src/dataset.rs

```diff
@@ -428,8 +432,10 @@ impl Dataset {
     ///
     /// See [`GDALClose`].
     ///
-    /// Note: on GDAL versions older than 3.7, this function always succeeds.
-    pub fn close(&mut self) -> Result<()> {
+    /// Note: on GDAL versions older than 3.7.0, this function always succeeds.
+    pub fn close(mut self) -> Result<()> {
```

Comment:
We could keep the old signature, but we'd have to check for `closed` on every public method. Or we could yank 0.15.0, which might be a good idea anyway.

Reply:
yes i think thats the way to go

Reply:
Which way? 😅

Reply:
change the signature

Reply:
oh sorry. I think we should change it to ` pub fn close(mut self) -> Result<()> `